tests/unit/models/UserInPermissionRoleTest.php

```diff
@@ -2,8 +2,82 @@
+        $userName = \Yii::app()->securityManager->generateRandomString(28);
+        $userIdSuperadminView = \User::insertUser($userName, $newPassword, 'John Doe', 1, $userName . '@example.org');
+        $userName = \Yii::app()->securityManager->generateRandomString(28);
+        $userIdSuperadminCreate = \User::insertUser($userName, $newPassword, 'John Doe', 1, $userName . '@example.org');
+
+        // User don't have any peermission
+        $this->assertFalse(\Permission::model()->hasGlobalPermission('auth_db', 'read', $userIdNoRoles));
+        $this->assertFalse(\Permission::model()->hasGlobalPermission('superadmin', 'read', $userIdNoRoles));
+
+        // Create a role superamin read
+        $newRoleName = \Yii::app()->securityManager->generateRandomString(12);
+        // Check if exist ? Near 0 possibility
+        $permissiontemplates = new \Permissiontemplates();
+        $permissiontemplates->name = $newRoleName;
+        $permissiontemplates->description = $newRoleName;
+        $permissiontemplates->renewed_last = date("Y-m-d");
+        $permissiontemplates->created_at = date("Y-m-d");
+        $permissiontemplates->created_by = 1;
+        $permissiontemplates->save();
+        $newRoleId = $permissiontemplates->ptid;
+
+        // Set the superadmin view roles
+        $permission = new \Permission();
+        $permission->entity = 'role';
+        $permission->entity_id = $newRoleId;
+        $permission->uid = 0;
+        $permission->permission = 'superadmin';
+        $permission->read_p = 1;
+        $permission->save();
+        $userInPermissionrole = new \UserInPermissionrole();
+        $userInPermissionrole->ptid = $newRoleId;
+        $userInPermissionrole->uid = $userIdSuperadminView;
+        $userInPermissionrole->save();
+        // User have all permission except create superadmin
+        $this->assertTrue(\Permission::model()->hasGlobalPermission('auth_db', 'read', $userIdSuperadminView));
+        $this->assertTrue(\Permission::model()->hasGlobalPermission('superadmin', 'read', $userIdSuperadminView));
+        $this->assertFalse(\Permission::model()->hasGlobalPermission('superadmin', 'create', $userIdSuperadminView));
+
+        // Add superadmin create
+        $permission->create_p = 1;
+        $permission->save();
+        $userInPermissionrole = new \UserInPermissionrole();
+        $userInPermissionrole->ptid = $newRoleId;
+        $userInPermissionrole->uid = $userIdSuperadminCreate;
+        $userInPermissionrole->save();
+        // User have create superadmin
+        $this->assertTrue(\Permission::model()->hasGlobalPermission('auth_db', 'read', $userIdSuperadminView));
+        $this->assertTrue(\Permission::model()->hasGlobalPermission('superadmin', 'read', $userIdSuperadminView));
+        $this->assertTrue(\Permission::model()->hasGlobalPermission('superadmin', 'create', $userIdSuperadminCreate));
+
```

Comment:
Maybe add forced superadmin here ?

Reply:
Seems interesting...
What would be the expected value to assert?